shuup_tests/default_tax/test_default_tax.py

```diff
@@ -237,6 +237,59 @@ def test_rule_admin(rf, admin_user):
     assert not TaxRule.objects.may_match_postal_code(postal_code).count()
 
 
+@pytest.mark.django_db
+def test_rules_with_anonymous():
+    """
+    Test the DefaultTaxModule with anonymous customer.
+    """
+
+    tax_class = TaxClass.objects.create(name="test")
+
+    # Create a product
+    shop = get_shop(prices_include_tax=False, currency='USD')
+    product = create_product('PROD', shop=shop, default_price=1000)
+    product.tax_class = tax_class
+    product.save()
+    price = product.get_shop_instance(shop).default_price
+
+    # create taxes
+    # When customer is company, it should pay additional taxes
+    tax_for_anyone = Tax.objects.create(code="any", rate=0.1, name="Tax for any customer")
+    tax_for_companies = Tax.objects.create(code="companies", rate=0.3, name="Additional tax for companies")
+
+    # create tax group for companies
+    companies_tax_group = CustomerTaxGroup.get_default_company_group()
+
+    # create the tax rule as follows:
+    # - 10% for any kind of customer, no matter what
+    # - 30% only for companies
+    any_tax_rule = TaxRule.objects.create(tax=tax_for_anyone)
+    any_tax_rule.tax_classes.add(tax_class)
```

Comment:
Can you give me a real life example for this? Like why and where anyone would do this? This sounds like a pricing module responsibility for me: "All anonymous customers pay 10% more". 

I have a really hard time to figure out why would any merchant do this and even harder time to figure out why we allow this.

Reply:
To be honest, I don't know any case here in my country for this.
But I have the certainty that me, as not inside any customer tax group, should be taxed the same as a customer tax group. What is the purpose of the `customer tax group` if not to limit?